frontend/vue/components/MiniComposer/CircuitSandboxWidget.vue

```diff
@@ -0,0 +1,276 @@
+      this.initialAvailableGates = this.stringToGateNameArray(availableGatesElement.textContent)
+    }
+  }
+
+  stringToGateNameArray (text: string) : ComposerGate[] {
+    let lastGateId = 0
+    return text.split(' ')
+      .filter(text => text !== '')
+      .map<ComposerGate>((gateText: string) => {
+        if (Object.values(GateName).some(gate => gateText.startsWith(gate))) {
+          const parts = gateText.split('(')
+          const gateName = parts[0]
+          const rotation = parts[1]?.split(')')[0]
+          return { name: gateName as GateName, id: lastGateId++, rotation }
+        }
+        return { name: GateName.UNKNOWN, id: lastGateId++ }
+      })
+  }
+
+  onRestartButton () {
+    this.circuitState = [[]]
+  }
+}
+</script>
+<style scoped lang="scss">
+@import 'carbon-components/scss/globals/scss/typography';
+@import 'carbon-components/scss/globals/scss/layout';
+@import '../../../scss/variables/colors.scss';
+@import '~/../scss/variables/mq.scss';
+
+.mini-composer {
+  display: grid;
+  grid-template-columns: 20rem 1fr;
+  grid-template-rows: min-content min-content 1fr min-content;
+  grid-template-areas:
+    "text text"
+    "gates lesson"
+    "circuit lesson"
+    "explanation lesson";
+
+  @include mq ($until: medium) {
+    grid-template-columns: 1fr;
+    grid-template-rows: repeat(5, min-content);
+    grid-template-areas:
+      "text"
+      "gates"
+      "circuit"
+      "explanation"
+      "lesson";
+  }
+
+  &__config-container {
+    display: none;
+  }
+
+  &__exercise-text {
+    @include type-style('body-long-01');
+    grid-area: text;
+    border-bottom: 1px solid $border-color;
+    padding: $spacing-05;
+  }
+  &__explanation {
+    @include type-style('body-long-01');
+    grid-area: explanation;
+    padding: $spacing-05;
+  }
+  &__gates {
+    grid-area: gates;
+    border-bottom: 1px solid $border-color;
+    padding: $spacing-05;
+    &__title {
+      margin: $spacing-03 0 $spacing-05 0;
+      @include type-style('heading-01');
+    }
+  }
+  &__circuit-section {
+    grid-area: circuit;
+    padding: $spacing-05;
+    &__title {
+      margin-top: $spacing-03;
+      @include type-style('heading-01');
+    }
+    &__qubit-line {
+      position: relative;
+      &__slot-container {
+        display: flex;
+        flex-flow: row;
+
+        position: absolute;
+        top: 0px;
+        height: 100%;
+        width: 100%;
+        align-items: center;
+        padding: 0 $spacing-07;
+      }
+      &__slot {
+        display: flex;
+        flex-flow: row;
+        flex: 1 1 auto;
+        max-width: 100%;
+
+        &__gate {
+          margin-right: $spacing-02;
+          &.sortable-ghost {
+            transition: opacity 0.2s ease-out;
+            opacity: 0.5;
+          }
+        }
+      }
+    }
+  }
+  &__state-views {
+    grid-area: lesson;
+    display: flex;
+    flex-direction: column;
+    gap: $spacing-05;
+    border-left: 1px solid $border-color;
```

Comment:
on mobile, I think we can hide this `border-left`
![Screen Shot 2022-06-13 at 11 31 09 AM](https://user-images.githubusercontent.com/6276074/173421796-1886e89c-4ce1-4f8a-baeb-8d4dc550654b.png)

Reply:
Also added a border top :)